### PR TITLE
tracing: update LightStep tracer

### PR DESCRIPTION
**Pull request by rnburn**
Description:

Updates to the a newer version of LightStep's tracer. Much of the tracer was rewritten to provide better performance.

In our [benchmarking](https://github.com/lightstep/lightstep-tracer-cpp/blob/8b7f871242a5d9f514a6eddf0ae51c15749d5ce7/benchmark/manual_tracer_benchmark.cpp), the speedup is around [2x-5x](https://7739-57146219-gh.circle-artifacts.com/0/benchmark/manual_tracer_benchmark_result.txt)

Risk Level: Medium
Testing:
Ran an [example](https://github.com/rnburn/envoy-lightstep-example) and verified that it reports traces.

### CODE REVIEW

**Comment by dio** [comment]
Seems like you need to merge master?

**Comment by moderation** [comment]
Does this fix https://github.com/envoyproxy/envoy/issues/7363 ?

File: bazel/repository_locations.bzl [hunk 1]
```diff
@@ -129,9 +129,9 @@ REPOSITORY_LOCATIONS = dict(
         urls = ["https://github.com/opentracing/opentracing-cpp/archive/v1.5.1.tar.gz"],
     ),
     com_lightstep_tracer_cpp = dict(
-        sha256 = "defbf471facfebde6523ca1177529b63784893662d4ef2c60db074be8aef0634",
-        strip_prefix = "lightstep-tracer-cpp-0.8.0",
-        urls = ["https://github.com/lightstep/lightstep-tracer-cpp/archive/v0.8.0.tar.gz"],
+        sha256 = "9dc52831b2ea5b9434bdd5d82850a6b24142e74b042d9bda8fa0532cdc8b4d51",
+        strip_prefix = "lightstep-tracer-cpp-66b88418ca5190eeb6131e2e021fdb7c2f3239ad",
+        urls = ["https://github.com/lightstep/lightstep-tracer-cpp/archive/66b88418ca5190eeb6131e2e021fdb7c2f3239ad.tar.gz"],
```
Comment (moderation) [inline bazel/repository_locations.bzl]: As you are picking up a commit after the last 0.12.0 release please add a comment date for your commit `# 2020-02-24`.

Comment (rnburn) [reply bazel/repository_locations.bzl]: Done

**Comment by rnburn** [comment]
Note: LightStep's tracer doesn't do anything with the gRPC response message. Only the response codes are relevant for determining if a report request succeeded.

Since the response isn't used, I changed the code to ignore it and dropped the invalid response tests.

**Comment by rnburn** [comment]
@lizan @moderation - could you guys take a look?

**Comment by moderation** [comment]
From a patch removal and dependency management POV it LGTM. Will need to defer to @lizan for the code changes

**Comment by moderation** [comment]
@rnburn I just built successfully for aarch64 but I get the following error when trying to run. Any tips on how to debug further and get more detailed output appreciated.
```
[2020-03-11 18:38:24.851][16423][info][tracing] [source/common/tracing/http_tracer_manager_impl.cc:25] instantiating tracing driver: envoy.tracers.lightstep
fish: '/home/username/Library/envoy/…' terminated by signal SIGILL (Illegal instruction)
```

**Comment by rnburn** [comment]
Does it have anything more in the stack trace?

**Comment by moderation** [comment]
I'm compiling with the following in my `.bazelrc` file. I'm guessing I'll need to change to get more info. I pinged you on Slack in case that might be faster:
```
build --define disable_object_dump_on_signal_trace=disabled
build --define signal_trace=disabled
```

**Pull request merged**